File: Data Structures & Algorithms/is-anagram/submission-0.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        sd={}
        for i in s:
            if i in sd:
                sd[i]+=1
            else:
                sd[i]=1
        for i in t:
            if i in sd:
                sd[i]-=1
            else:
                print('a')
                return False
        print(sd)
        for i in sd:
            if sd[i]!=0:
                print('b')
                return False
        return True



        
        

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        
        HashMap<Integer, Integer> map=new HashMap<>();
        for(int i: nums) map.merge(i, 1, Integer::sum);   //(a,b)->Integer.sum(a,b)
        System.out.println(map);
        List<List<Integer>> list=new ArrayList<>();
        for(int i=0; i<=nums.length; i++) list.add(new ArrayList<Integer>());
        for(Integer i: map.keySet()){
            int idx=map.get(i);
            System.out.println(idx);
            list.get(idx).add(i);
        }
        int[] res=new int[k];
        int count=0, flag=0;
        for(int i=nums.length; i>=0; i--){
            for(int j=0; j<list.get(i).size(); j++){
                res[count++]=list.get(i).get(j);
                if(count==k){ flag=1; break; }
            }
            if(flag==1) break;
        }
        return res;
    }
}
